instaloader/structures.py

```diff
@@ -373,14 +375,20 @@ def is_video(self) -> bool:
     @property
     def video_url(self) -> Optional[str]:
         """URL of the video, or None."""
+        def get_size(url):
+            return requests.head(url).headers.get('Content-Length', 0), url
         if self.is_video:
+            url_candidates = []
+            if 'video_url' in self._node:
+                url_candidates.append(get_size(self._node['video_url']))
             if self._context.is_logged_in:
                 try:
-                    url = self._iphone_struct['video_versions'][0]['url']
-                    return url
+                    for version in self._iphone_struct['video_versions']:
+                            url_candidates.append(get_size(version['url']))
```

Comment:
```suggestion
                        url_candidates.append(get_size(version['url']))
```

To fix Pylint's `bad-indentation` warning.

Other than that, it looks good I think.